server-spi-private/src/main/java/org/keycloak/userprofile/AttributeGroupMetadata.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.userprofile;
+
+import java.util.Map;
+
+/**
+ * Configuration of the attribute group.
+ *
+ * @author <a href="joerg.matysiak@bosch.io">Jörg Matysiak</a>
+ */
+public class AttributeGroupMetadata {
```

Comment:
Do we really need all this completely separate and complex handling of groups? For me they are only additional info for GUI rendering. Much easier/simpler would be to keep complete group info in `AttributeMetdata` of individual attribute, and let `org.keycloak.userprofile.DeclarativeUserProfileProvider.decorateUserProfileForCache(UserProfileMetadata, ComponentModel)` to set it into each AttributeMetadata based on parsed configuration.

Reply:
ok, since @pedroigor mention something similar above I'm going to adapt the code, will push an updated version probably on monday

Reply:
thanks, I think it will be much simpler code

Reply:
thanks

Reply:
@velias  @pedroigor, I pushed an update that provides group metadata as part of the attribute metadata